src/machine.rs

```diff
@@ -1572,3 +1578,19 @@ pub(crate) async fn promise_result_to_future(
         }
     }
 }
+
+async fn stream_to_json_array<T, S>(mut stream: Pin<&mut S>) -> Result<String, anyhow::Error>
+where
+    T: Serialize,
+    S: Stream<Item = T>,
+{
+    let mut stream_json = vec![];
```

Comment:
Would initializing with an initial capacity make sense? Assuming the stream's not empty, we can likely avoid many reallocations by having a large initial size here?

Reply:
I think so, but I don't know how to estimate a good size here. Do we do anything similar anywhere else in the code? Do we have any rules of thumb?